src/site/content/en/metrics/inp/index.md

```diff
@@ -110,14 +110,19 @@ Since INP is experimental, the guidance around thresholds may change over time a
 
 The primary driver of interactivity is often JavaScript, though browsers do provide interactivity through controls _not_ powered by JavaScript, such as checkboxes, radio buttons, and controls powered by CSS.
 
-As far as INP goes, **only the following interaction types are observed:**
+As far as INP goes, **only the following _synchronous_ interaction types are observed:**
```

Comment:
I think adding the word synchronous to describe interaction types is confusing. Interactions themselves aren't really defined as synchronous or asynchronous; instead some of the work they kick off is synchronous and some of it asynchronous.

For example, let's go back to the tweet thread with the two codepens.

In [Adriano's example](https://codepen.io/isadrisal/pen/abYVVGd), there *is* some synchronous work, just not very much:

```
  const target = evt.target;
  if (target.tagName !== 'BUTTON') return;
  
  const originalText = target.textContent;
  const delay = target.dataset.slow ? 2000 : 0;
  
  setTimeout(()
```

That will be measured for INP, and we'll generally get a score of 0 because there is so little work. But it's still measured. The rest of the work is asynchronous, because it doesn't need to happen in the event handler or in order to display a frame:

```
    target.textContent = 'Clicked';
    target.classList.add('clicked');
    setTimeout(() => {
      target.textContent = originalText;
      target.classList.remove('clicked');
    }, 400);
```

If we waited for _all_ the asynchronous work to complete, we'd wait 2000ms for the first setTimeout, then ~0ms for the code to run, then 400ms for the second setTimeout, and then ~16ms for another frame. And that would be really unfair, because we'd be counting not just the time to update the UI, but also the animation to un-highlight the button. And that's one reason we don't currently count asynchronous work; it's very hard to understand what the user is or isn't waiting for.

In [my example](https://codepen.io/anniesullie/pen/eYMeQBo) there is also synchronous work:

```
  const target = evt.target;
  if (target.tagName !== 'BUTTON') return;
  
  const originalText = target.textContent;
  const delay = target.dataset.slow ? 2000 : 0;
  
  const startTime = performance.now();
  let currentTime = performance.now();
  while (currentTime - startTime < delay) {
    currentTime = performance.now();
  }
  
  target.textContent = 'Clicked';
  target.classList.add('clicked');
  setTimeout(() 
```

This work takes much longer and blocks the main thread because of the busy loop. Since I put it in the event handler, it needs to run before the main thread for the page can do other work, so the UI is locked for 2 seconds. And I get a resulting bad INP score of 2016ms (the 2 seconds plus the time to update the UI)

But my version _also_ does some asynchronous work, this code inside the setTimeout:
```
    target.textContent = originalText;
    target.classList.remove('clicked');
```

This additional wait for 400ms to run this code and the time to run the code and produce a frame won't be counted towards INP.

Reply:
To clarify, my feedback would be to not change line 113.

Reply:
I agree with not changing this line.

I think the place to mention that it's only sync/blocking work that INP considers is either in the "What is INP?" or "What's in an interaction?" sections. Both of those sections break down INP and define the terms "interaction latency" and "event duration", and I think that's where we want to clarify that any work that gets scheduled/postpone to run in a future frame will not contribute to INP.

Also (meta point), I'm not 100% sure we want to refer to this a sync vs. async work, since microtasks are async but they can and do block the main thread.

Reply:
I think the aside I placed later on in this section clarifies this sufficiently, @philipwalton, but let me know what you think. The changes are on lines 120-124.